lib/src/textboxGroup/textboxGroup.component.ts

```diff
@@ -0,0 +1,153 @@
+import {
+    Component,
+    Input,
+    ViewEncapsulation,
+    ElementRef,
+    forwardRef,
+    Provider,
+    OnChanges,
+    SimpleChanges,
+    OnInit,
+    Output,
+    EventEmitter,
+} from "@angular/core";
+import { randomId } from "@sebgroup/frontend-tools/dist/randomId";
+
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+
+const CUSTOM_DROPDOWN_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => TextboxGroupComponent),
+    multi: true,
+};
+
+type Value = number | string;
+
+@Component({
+    selector: "sebng-textboxgroup",
+    templateUrl: "./textboxGroup.component.html",
+    styleUrls: ["./textboxGroup.component.scss"],
+    encapsulation: ViewEncapsulation.None,
+    providers: [CUSTOM_DROPDOWN_CONTROL_VALUE_ACCESSOR],
+})
+export class TextboxGroupComponent implements ControlValueAccessor, OnInit, OnChanges {
+    @Input() autoComplete?: "on" | "off";
+    @Input() className?: string;
+    @Input() disabled?: boolean;
+    @Input() error?: string;
+    @Input() focus?: boolean;
+    @Input() id?: string;
+    @Input() label?: string;
+    @Input() leftIcon?: string;
+    @Input() leftText?: string;
+    @Input() leftTitle?: string;
+    @Input() maxLength?: number;
+    @Input() minLength?: number;
+    @Input() name: string;
+    @Input() pattern?: string;
+    @Input() placeholder?: string;
+    @Input() readOnly?: boolean;
+    @Input() required?: boolean;
+    @Input() rightIcon?: string;
+    @Input() rightText?: string;
+    @Input() rightTitle?: string;
+    @Input() type?: string;
+    @Input() success?: boolean;
+    @Input() showErrorMessage?: boolean;
+
+    @Output() onBlur = new EventEmitter<MouseEvent>();
+    @Output() onChange = new EventEmitter<Value>();
+    @Output() onFocus = new EventEmitter<MouseEvent>();
+    @Output() onKeyDown = new EventEmitter<KeyboardEvent>();
+    @Output() onKeyPress = new EventEmitter<KeyboardEvent>();
+    @Output() onKeyUp = new EventEmitter<KeyboardEvent>();
+    @Output() onLeftClick = new EventEmitter<MouseEvent>();
+    @Output() onRightClick = new EventEmitter<MouseEvent>();
+
+    private innerValue: Value;
+
+    private onTouchedCallback: () => void;
+    private onChangeCallback: (_: any) => void;
+
+    public internalId: string;
+    public displayError: boolean;
```

Comment:
same here

Reply:
They are not privates , we use them within the .html file.

Reply:
These are not inputs and should not be exposed to the users, `private` would be a better declaration

Reply:
No they are not private, if you make them private , then you can't access them outside the file especially inside the .html file.